File: 932-monotonic-array/monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        bool increasing = true;
        bool decreasing = true;

        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] < nums[i + 1]) {
                decreasing = false;  // saw an increase
            }
            if (nums[i] > nums[i + 1]) {
                increasing = false;  // saw a decrease
            }
        }

        return increasing || decreasing;
    }
};

File: 932-monotonic-array/monotonic-array.py
class Solution:
    def isMonotonic(self, nums: list[int]) -> bool:
        increasing = True
        decreasing = True

        for i in range(len(nums) - 1):
            if nums[i] < nums[i + 1]:
                decreasing = False  # we saw an increase, can't be decreasing
            if nums[i] > nums[i + 1]:
                increasing = False  # we saw a decrease, can't be increasing

        return increasing or decreasing